src/main/kotlin/io/sip3/salto/ce/socket/ManagmentSocket.kt

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.salto.ce.socket
+
+import io.sip3.commons.domain.SdpSession
+import io.sip3.salto.ce.RoutesCE
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.datagram.DatagramSocket
+import io.vertx.core.json.JsonObject
+import io.vertx.core.net.SocketAddress
+import io.vertx.ext.mongo.MongoClient
+import io.vertx.kotlin.ext.mongo.updateOptionsOf
+import mu.KotlinLogging
+import java.net.URI
+
+/**
+ * Management socket
+ */
+class ManagementSocket : AbstractVerticle() {
+
+    private val logger = KotlinLogging.logger {}
+
+    companion object {
+
+        const val TYPE_SDP_SESSION = "sdp_session"
+        const val TYPE_REGISTER = "register"
+    }
+
+    private var client: MongoClient? = null
+
+    private lateinit var uri: URI
+    private var expirationTimeout: Long = 120000
+    private var expirationDelay: Long = 60000
+
+    private val remoteHosts = mutableMapOf<String, RemoteHost>()
+    lateinit var socket: DatagramSocket
+
+    override fun start() {
+        config().getJsonObject("mongo")?.let { config ->
+            client = MongoClient.createShared(vertx, JsonObject().apply {
+                put("connection_string", config.getString("uri") ?: throw IllegalArgumentException("mongo.uri"))
+                put("db_name", config.getString("db") ?: throw IllegalArgumentException("mongo.db"))
+            })
+        }
+
+        config().getJsonObject("management").let { config ->
+            uri = URI(config.getString("uri") ?: throw IllegalArgumentException("uri"))
+            config.getLong("expiration-timeout")?.let { expirationTimeout = it }
+            config.getLong("expiration-delay")?.let { expirationDelay = it }
+        }
+
+        startUdpServer()
+
+        vertx.eventBus().localConsumer<List<SdpSession>>(RoutesCE.sdp_session) { event ->
+            val sdpSessions = event.body()
+            sdpSessions.forEach { publishSdpSession(it) }
+        }
+
+        vertx.setPeriodic(expirationDelay) {
+            val now = System.currentTimeMillis()
+            remoteHosts.filterValues { it.lastUpdate + expirationTimeout < now }
+                    .forEach { remoteHosts.remove(it.key) }
+        }
+    }
+
+    private fun publishSdpSession(sdpSession: SdpSession) {
+        val buffer = JsonObject().apply {
+            put("type", TYPE_SDP_SESSION)
+            put("payload", JsonObject.mapFrom(sdpSession))
+        }.toBuffer()
+
+        remoteHosts.values.forEach { remoteHost ->
+            try {
+                socket.send(buffer, remoteHost.uri.port, remoteHost.uri.host) {}
+            } catch (e: Exception) {
+                logger.error("Failed to send to ${remoteHost.uri}: ${e.message}", e)
```

Comment:
Let's format error message in accordance with typical structure. `Something `something()` failed. ...`